clippy_lints/src/utils/conf.rs

```diff
@@ -284,6 +284,12 @@ define_Conf! {
     ///
     /// The list of unicode scripts allowed to be used in the scope.
     (allowed_scripts: Vec<String> = vec!["Latin".to_string()]),
+    /// Lint: INDEX_REFUTABLE_SLICE.
+    ///
+    /// The limit after which slice indexing will not be linted as avoidable.
+    ///
+    /// As an example, the default limit of `3` would allow `slice[3]` but lint `slice[2]`.
```

Comment:
I'm not quite sure how to create a generic configuration documentation for this. :sweat_smile:  I've updated the comment now to say:

```rs
    /// The limit after which slice indexing will not be linted.
    ///
    /// As an example, the default limit of `3` would allow `slice[3]` but lint `slice[2]`.
```

Reply:
Something like

> When Clippy suggests using a slice pattern, this is the maximum number of elements allowed in the slice pattern that is suggested. If more elements would be necessary, the lint is suppressed. For example, `[_, _, _, e, ..]` is a slice pattern with 4 elements.

Here I'm implying that `..` doesn't count. That's consistent with the current impl right? But maybe it should count? I don't have a strong opinion on that.